express/src/validators/hiddenNumberMobileOtpValidator.ts

```diff
@@ -0,0 +1,39 @@
+import {NextFunction, Request, Response} from "express";
+import {obscureNumber} from "../lib/mobileNumberUtils";
+import {validateOtp} from "../lib/checkOtp";
+
+type MiddlewareFunction<T, U, V> = (T: Request, U: Response, V: NextFunction) => void;
+
+export function mobileOtpValidator(
+    formActionUrl: string,
+    textMessageNotReceivedUrl: string
+): MiddlewareFunction<Request, Response, NextFunction> {
+    return async (req: Request, res: Response, next: NextFunction) => {
+        let otp: string = req.body["sms-otp"];
+        otp = otp.trim();
+        const mobileNumber = obscureNumber(getMobileNumber(req));
```

Comment:
There's too much duplication between the obscuring and non-obscuring validators. It's fine to have a boolean to obscure the number in the function args (with a default value of false, say). Or do the obscuring outside the validator instead, or have a common util/class that can be reused. I'd probably just go with the boolean.